packages/common/src/ide/inMemoryTextDocument/InMemoryTextDocument.test.ts

```diff
@@ -0,0 +1,127 @@
+import * as assert from "node:assert";
+import { URI } from "vscode-uri";
+import { Position } from "../../types/Position";
+import { InMemoryTextDocument } from "./InMemoryTextDocument";
+
+const uri = URI.parse("cursorless-dummy://dummy/untitled");
+const languageId = "plaintext";
+
+suite("InMemoryTextDocument", () => {
+  test("constructor", () => {
+    const document = new InMemoryTextDocument(uri, languageId, "hello\nworld");
+
+    assert.equal(document.uri.toString(), "cursorless-dummy://dummy/untitled");
+    assert.equal(document.languageId, "plaintext");
+    assert.equal(document.filename, "untitled");
+    assert.equal(document.text, "hello\nworld");
+    assert.equal(document.lineCount, 2);
+    assert.equal(document.eol, "LF");
+    assert.equal(document.version, 0);
+  });
+
+  test("filename", () => {
+    const document1 = new InMemoryTextDocument(
+      URI.parse("cursorless-dummy://dummy/foo.ts"),
+      "",
+      "",
+    );
+    const document2 = new InMemoryTextDocument(
+      URI.file("dummy\\bar.ts"),
+      "",
+      "",
+    );
+
+    assert.equal(document1.filename, "foo.ts");
+    assert.equal(document2.filename, "bar.ts");
+  });
+
+  test("CRLF", () => {
+    const document = new InMemoryTextDocument(
+      uri,
+      languageId,
+      "foo\nbar\r\nbaz",
+    );
+
+    assert.equal(document.lineCount, 3);
+    assert.equal(document.eol, "CRLF");
+  });
+
+  test("lineAt", () => {
```

Comment:
Would be good to have some edge cases. At least these doc content examples:

- `""`
- `"aaa"`
- `"\naaa"`
- `"aaa\n"`
- `"\naaa\n"`
- `"\n"`
- `"\n\n"`
- `"aaa\n\naaa"`
- `"aaa\nbbb\r\nccc"`

You don't need to test all attributes of the lines for these; I think just range and range including newline? Worth doing the usual fixture thing with a TestCase interface containing input and expected output to make these more readable. Just short repr of the expected output ranges is fine I think

Reply:
Tests added